lime/lime_tabular.py

```diff
@@ -363,6 +372,9 @@ def explain_instance(self,
         if feature_names is None:
             feature_names = [str(x) for x in range(data_row.shape[0])]
 
+        if sp.sparse.issparse(data_row):
+            # Note: need to optimize this more to remove flatten
+            data_row = data_row.toarray().flatten()
```

Comment:
need to remove flatten here and fix the rest of the code below to handle sparse data

Reply:
the TableDomainMapper densifies the row in order to show it as html.  Not sure how much I should change here.  In sparse case, we would probably want to show the non-zero indexes and values instead of values as a list in the visualization.

Reply:
done, only display non-zeros in sparse case